From bc4a0d9a2abfcd75d9b0ee4be3f7600905fe6001 Mon Sep 17 00:00:00 2001
From: Marek Kasik <mkasik@redhat.com>
Date: Fri, 24 Jun 2022 22:01:27 +0200
Subject: [PATCH 2/2] Form: Provide Unicode marker when ensuring fonts

Form::ensureFontsForAllCharacters() needs input text with Unicode marker.
Provide such in FormFieldText::setContentCopy().
---
 poppler/Form.cc | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/poppler/Form.cc b/poppler/Form.cc
index d33160b3..9407b6c5 100644
--- a/poppler/Form.cc
+++ b/poppler/Form.cc
@@ -27,7 +27,7 @@
 // Copyright 2019, 2020 Oliver Sander <oliver.sander@tu-dresden.de>
 // Copyright 2019 Tomoyuki Kubota <himajin100000@gmail.com>
 // Copyright 2019 João Netto <joaonetto901@gmail.com>
-// Copyright 2020, 2021 Marek Kasik <mkasik@redhat.com>
+// Copyright 2020-2022 Marek Kasik <mkasik@redhat.com>
 // Copyright 2020 Thorsten Behrens <Thorsten.Behrens@CIB.de>
 // Copyright 2020 Klarälvdalens Datakonsult AB, a KDAB Group company, <info@kdab.com>. Work sponsored by Technische Universität Dresden
 // Copyright 2021 Georgiy Sgibnev <georgiy@sgibnev.com>. Work sponsored by lab50.net.
@@ -1674,14 +1674,14 @@ void FormFieldText::setContentCopy(const GooString *new_content)
                     Object fieldResourcesDictObj = obj.dictLookup("DR");
                     if (fieldResourcesDictObj.isDict()) {
                         GfxResources fieldResources(doc->getXRef(), fieldResourcesDictObj.getDict(), form->getDefaultResources());
-                        const std::vector<Form::AddFontResult> newFonts = form->ensureFontsForAllCharacters(new_content, fontName, &fieldResources);
+                        const std::vector<Form::AddFontResult> newFonts = form->ensureFontsForAllCharacters(content, fontName, &fieldResources);
                         // If we added new fonts to the Form object default resuources we also need to add them (we only add the ref so this is cheap)
                         // to the field DR dictionary
                         for (const Form::AddFontResult &afr : newFonts) {
                             fieldResourcesDictObj.dictLookup("Font").dictAdd(afr.fontName.c_str(), Object(afr.ref));
                         }
                     } else {
-                        form->ensureFontsForAllCharacters(new_content, fontName);
+                        form->ensureFontsForAllCharacters(content, fontName);
                     }
                 }
             } else {
-- 
2.38.1

